Python/pytime.c

```diff
@@ -294,7 +301,19 @@ pytime_double_to_denominator(double d, time_t *sec, long *numerator,
     }
     assert(0.0 <= floatpart && floatpart < denominator);
 
-    if (!_Py_InIntegralTypeRange(time_t, intpart)) {
+    /*
+       Conversion of an out-of-range value to time_t gives undefined behaviour
+       (C99 §6.3.1.4p1), so we must guard against it. However, checking that
+       `intpart` is in range is delicate: the obvious expression `intpart <=
+       PY_TIME_T_MAX` will first convert the value `PY_TIME_T_MAX` to a double,
+       potentially changing its value and leading to us failing to catch some
+       UB-inducing values. The code below works correctly under the mild
+       assumption that time_t is a two's complement integer type with no trap
+       representation, and that `PY_TIME_T_MIN` is within the representable
+       range of a C double.
+    */
+    assert(fmod(intpart, 1.0) == 0.0);
+    if (!((double)PY_TIME_T_MIN <= intpart && intpart < -(double)PY_TIME_T_MIN)) {
```

Comment:
Note: it may be tempting to suggest applying De Morgan's laws here to remove the negation, but that would do the wrong thing with NaNs. (It's not immediately clear to me whether NaNs could make it through to this bit of code, but better safe than sorry.)

Reply:
I suggest adding this into the comment for future reference.

Reply:
Done.